### Homework 2

### Question 1: Comparing Two Population Means (Independent t-test)

### Hypotheses

- Null hypothesis ($H_0$): The mean increase in sleep is the same for both drugs, $\mu_1 - \mu_2 = 0$.
- Alternative hypothesis ($H_A$): The mean increase in sleep differs between the drugs, $\mu_1 - \mu_2 \ne 0$.

Because the question asks whether the means differ, this is a two-sided independent-samples t-test. Welch's version is used because it does not assume equal population variances.

The significance level is $\alpha = 0.05$.

In [ ]:
import numpy as np
from scipy import stats


sleep_drug_1 = np.array([0.7, -1.6, -0.2, -1.2, -0.1, 3.4, 3.7, 0.8, 0.0, 2.0])
sleep_drug_2 = np.array([1.9, 0.8, 1.1, 0.1, -0.1, 4.4, 5.5, 1.6, 4.6, 3.4])

alpha = 0.05
result = stats.ttest_ind(sleep_drug_1, sleep_drug_2, equal_var=False)
mean_difference = sleep_drug_1.mean() - sleep_drug_2.mean()

print(f"Drug 1 mean: {sleep_drug_1.mean():.2f} hours")
print(f"Drug 2 mean: {sleep_drug_2.mean():.2f} hours")
print(f"Mean difference (drug 1 - drug 2): {mean_difference:.2f} hours")
print(f"Welch t-statistic: {result.statistic:.3f}")
print(f"Two-sided p-value: {result.pvalue:.4f}")

if result.pvalue < alpha:
    print("Decision: Reject H0. The mean increases differ significantly.")
else:
    print("Decision: Fail to reject H0. There is not enough evidence of a significant difference.")

Drug 1 mean: 0.75 hours
Drug 2 mean: 2.33 hours
Mean difference (drug 1 - drug 2): -1.58 hours
Welch t-statistic: -1.861
Two-sided p-value: 0.0794
Decision: Fail to reject H0. There is not enough evidence of a significant difference.


### Results and conclusion

- The sample mean increase was 0.75 hours for drug 1 and 2.33 hours for drug 2. 
- Thus, drug 2 had a sample mean increase that was 1.58 hours larger than drug 1.

The Welch independent t-test produced $t = -1.861$ and a two-sided $p$-value of 0.0794. Since $p > 0.05$, we fail to reject $H_0$. There is not sufficient statistical evidence at the 5% significance level to conclude that the mean increase in sleep differs between the two drugs.

This conclusion does not prove that the drugs have identical effects; it means that this sample does not provide strong enough evidence of a difference at $\alpha = 0.05$.